src/openvic-simulation/misc/OnActions.cpp

```diff
@@ -0,0 +1,37 @@
+#include "OnActions.hpp"
+
+using namespace OpenVic;
+using namespace OpenVic::NodeTools;
+
+OnAction::OnAction(std::string_view new_identifier, weight_map_t new_weighted_events)
+	: HasIdentifier { new_identifier }, weighted_events { std::move(new_weighted_events) } {}
+
+bool OnActionManager::add_on_action(std::string_view identifier, OnAction::weight_map_t weighted_events) {
+	if (identifier.empty()) {
+		Logger::error("Invalid decision identifier - empty!");
+		return false;
+	}
+
+	return on_actions.add_item({ identifier, std::move(weighted_events) });
+}
+
+bool OnActionManager::load_on_action_file(EventManager const& event_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary([this, &event_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		OnAction::weight_map_t weighted_events;
+		bool ret = expect_dictionary(
+			[&identifier, &weighted_events, &event_manager](std::string_view weight_str, ast::NodeCPtr event_node) -> bool {
+				std::string_view event_id;
+				uint64_t weight = StringUtils::string_to_uint64(std::string(weight_str));
+				bool ret = expect_identifier_or_string(assign_variable_callback(event_id))(event_node);
```

Comment:
```suggestion
				Event const* event = nullptr;
				bool ret = event_manager.expect_event_identifier(assign_variable_callback_pointer(event))(event_node);
```
The `Event const*` can be got directly from the node.